pkg/scheduler/extender/extender.go

```diff
@@ -252,7 +250,20 @@ func (e *Extender) gatherCapacityRequestsByProvisioner(ctx context.Context, pod
 		}
 		if v.Ephemeral != nil {
 			claimSpec := v.Ephemeral.VolumeClaimTemplate.Spec
-			if storageType, ok := scs[*claimSpec.StorageClassName]; ok {
+			if claimSpec.StorageClassName == nil {
+				ll.Warningf("PVC %s skipped due to empty StorageClass", v.Ephemeral.VolumeClaimTemplate.Name)
+				continue
+			}
+
+			storageType, scType := scChecker.check(*claimSpec.StorageClassName)
+			switch scType {
```

Comment:
Can we `DRY` those switches, the code from 259-278 and 312-332 looks like the same.

Reply:
We need to have different log messages and request fields for them. In my opinion, creating a common interface for just 2 calls brings more code here

Reply:
But you can just create an anonymous function with all repeated stuff, less code is better that the same code repeated two times.

Reply:
But it is used `return` and `continue` in switch cases. We should whatever return a status code and parse it in another switch in the main loop. Or what do you mean?

Reply:
Could you please suggest another design for `scChecker` component to discuss? 

In my opinion, we return the status code from the function and just execute 2 strings - print log and make a decision about loop. In the third case `relatedSC` execution is different, so creating common function leads to more code due to interface usage and a lot of parameters.

I agree that making functions for repeatable code is a good practice and must be used as much as possible. But, from my perspective, in this case, it's not applicable.

We can add a new return code to the enum in the future and will have to change `switch`es too. So `default` section helps to simplify error locating for developer during maintaining this code. I can place `panic` here, but it's not a kubernetes way due to log corrupting

Reply:
> Could you please suggest another design for scChecker component to discuss?

Something simpler.

> ‎creating common function leads to more code due to interface usage and a lot of parameters.
DRY is not about more code, is about less code.

> We can add a new return code to the enum in the future

[Stop Predicting the Future of Your Code](https://medium.com/swlh/stop-predicting-the-future-of-your-code-e0d403116f90)

Reply:
May be it make sense to move switch statement to separate function.